nixos/modules/services/misc/nocodb.nix

```diff
@@ -0,0 +1,60 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.nocodb;
+
+  defaultEnvs = {
+    DATABASE_URL = "sqlite:///%S/nocodb/sqlite.db";
+  };
+in
+{
+  meta.maintainers = with lib.maintainers; [ sinanmohd ];
+
+  options.services.nocodb = {
+    enable = lib.mkEnableOption "nocodb";
+    package = lib.mkPackageOption pkgs "nocodb" { };
+
```

Comment:
A little detail that I think it would be a nice touch:

You can set which port the service will listen on by passing it to the PORT variable.